### PR TITLE
Relational data model - clean version

### PR DESCRIPTION
**Pull request by sknick-iastate**
Based on [@j-d-b](https://github.com/usdot-jpo-ode/jpo-wzdx/pull/50#issuecomment-555106423) comments, this pull request is a clean version of the relational data model that @DeraldDudley has developed without the additional changes from other pull requests.  This pull request only changes the structure of the data from version 1.1 to a relational model and very little of the content. All other changes in Pull Request #50 should be made through separate pull requests.

**_A summary of some of the proposed changes:_**
-[Logical Data Model ](https://github.com/sknick-iastate/jpo-wzdx/blob/RelationalDataModel/data-tables/road_event_erd.png)(Entity Relationship Data Model)
-Renamed [WZDx Header](https://github.com/usdot-jpo-ode/jpo-wzdx/blob/master/data-tables/wzdx-header-data.md) to [road_event_feed_info](https://github.com/sknick-iastate/jpo-wzdx/blob/RelationalDataModel/data-tables/road_event_feed_info.md)
-Renamed [Common Core](https://github.com/usdot-jpo-ode/jpo-wzdx/blob/master/data-tables/common-core-dictionary.md) to [road_events](https://github.com/sknick-iastate/jpo-wzdx/blob/RelationalDataModel/data-tables/road_events.md)
-Add geometry_type and geometry to replace the begin and end location data frames. Also moved other elements within those data frames to the road_events table.
-Added data elements from the start and end time data frames to the road_events table
-Removed the estimated and verified tags and replaced those with begin location, end location, start time and end time accuracy fields.
-Update the [examples](https://github.com/sknick-iastate/jpo-wzdx/tree/RelationalDataModel/create-feed) in the create-feed folder (would be beneficial to have some one review this in detail)
-Removed the [full_spec](https://github.com/usdot-jpo-ode/jpo-wzdx/tree/master/full-spec) since is was redundant and could result in discrepancies.  All content within the full_spec that was not located within github was added to the top README.md file.
-Update the [sample files](https://github.com/sknick-iastate/jpo-wzdx/tree/RelationalDataModel/sample-files) including xml, json, and xsd

I also addressed some of the comments related to [@j-d-b](https://github.com/usdot-jpo-ode/jpo-wzdx/pull/50#issuecomment-555282214) which I will summarize in the first comment.

**_Why_**: Version 1 of the Work Zone Data Specification relies on data dictionary, markup, and exchange languages (TMDD, XML, JSON) to define content and formatting. It does not employ well established data design methods to define objects, object properties, or relationships.

Using Relational Data Modeling Techniques to design the Work Zone Data Specification will;

- Organize the data more efficiently
- Increase consistency
- Ease data retrieval
- Enable a flexible and scalable data model
- Promote Standards-Centric development of applications

Using Relational Data Modeling Techniques to design the Work Zone Data Specification will not;

- Dictate how data is managed at the source
- Change the specifications content
- Prohibit publishing content according to the TMDD
- Prohibit publishing content according to version 1 of the specification

### CODE REVIEW

**Comment by sknick-iastate** [comment]
@j-d-b I tried to address some of your [actions/comments/questions](https://github.com/usdot-jpo-ode/jpo-wzdx/pull/50#issuecomment-555282214) related to the relational model from last night on #50 which I have summarized below.

**Needs Action**
- third bullet point, I agreed on removing the `full-spec` which has been removed.  It resulted in some discrepancies that could be avoided since the data is located in other directories within the repository.  Anything that I could not find in any of the other repositories was added to the bottom of the top level [README.md](https://github.com/sknick-iastate/jpo-wzdx/blob/RelationalDataModel/README.md) file
- fourth bullet point, I have updated the `create-feed` directory but it probably needs a good review especially the JSON samples
-fifth bullet point, the `sample-files` directory still needs updated
-sixth bullet point, I updated the project root README.md (https://github.com/sknick-iastate/jpo-wzdx/blob/RelationalDataModel/README.md) file to remove references to common core.

**Comments**
- first bullet point, not addressed here but could be with a separate PR as you mentioned
-second bullet point, I removed the milepost demarcation since I feel they are redundant of the begin/end location demarcation (which I renamed to accuracy).  Happy to add this back if other feel it is needed.
-fourth bullet point, this was one issue I had with PR #50 as well.  I have kept the issuingOrganization in the road_events table
-fifth bullet point, I added the hyperlinks 

**Questions**
-first bullet point, I'm not sure of the source but changed it to accuracy since that seems like a better definition.  Can easily change if desired
-second bullet point, I could go either way on this.  I think `event_direction` makes sense since it is the direction of the work zone closure.  But I can also see the argument for `road_direction`.  Happy to change if needed
-I agree a migration guide would be beneficial.  I think it would be good to hold off until all of the version 2 changes are made

**Comment by j-d-b** [comment]
> @j-d-b I tried to address some of your actions/comments/questions related to the relational model from last night on #50 which I have summarized below.

@sknick-iastate thank you for responding to some of these and for making a clean PR without changes. As for responding to some of my actions/comments/questions, would you be able to do a quoted reply of my full list here and comment on each in line? I think it would be helpful for myself and others to see which have been addressed and which have no comment without having to jump back and forth between threads and check your commits here.

**Comment by DeraldDudley** [comment]
> @j-d-b I tried to address some of your [actions/comments/questions](https://github.com/usdot-jpo-ode/jpo-wzdx/pull/50#issuecomment-555282214) related to the relational model from last night on #50 which I have summarized below.
> 
> **Needs Action**
> 
> * third bullet point, I agreed on removing the `full-spec` which has been removed.  It resulted in some discrepancies that could be avoided since the data is located in other directories within the repository.  Anything that I could not find in any of the other repositories was added to the bottom of the top level [README.md](https://github.com/sknick-iastate/jpo-wzdx/blob/RelationalDataModel/README.md) file
> * fourth bullet point, I have updated the `create-feed` directory but it probably needs a good review especially the JSON samples
>   -fifth bullet point, the `sample-files` directory still needs updated
>   -sixth bullet point, I updated the project root README.md (https://github.com/sknick-iastate/jpo-wzdx/blob/RelationalDataModel/README.md) file to remove references to common core.
> 
> **Comments**
> 
> * first bullet point, not addressed here but could be with a separate PR as you mentioned
>   -second bullet point, I removed the milepost demarcation since I feel they are redundant of the begin/end location demarcation (which I renamed to accuracy).  Happy to add this back if other feel it is needed.

>   -fourth bullet point, this was one issue I had with PR #50 as well.  I have kept the issuingOrganization in the road_events table
>   -fifth bullet point, I added the hyperlinks
> 
> **Questions**
> -first bullet point, I'm not sure of the source but changed it to accuracy since that seems like a better definition. Can easily change if desired
> -second bullet point, I could go either way on this. I think `event_direction` makes sense since it is the direction of the work zone closure. But I can also see the argument for `road_direction`. Happy to change if needed
> -I agree a migration guide would be beneficial. I think it would be good to hold off until all of the version 2 changes are made

@sknick-iastate & @j-d-b 
Thanks for your work on this.  

Some thoughts.
1. If we're normalizing the data, the issuingOrganization should be stored in road_event_info_feed table.  Is there a compelling reason to list the issuing organization for every event when listing it for the feed conveys the same information more efficiently?

2. I changed the accuracy descriptor because the information being conveyed is not a measure of accuracy; Its an indication of whether or not the provided information have been verified.  How do you feel about using the is_verified flag for "accuracy" fields.
- event_beginning_is_verified
- event_ending_is_verified 
- event_start_date_is_verified
- event_end_date_is_verified
- event_beginning_milepost_is_verified
- event_ending_milepost_is_verified

3. I used event_direction because of the business rule that work zones gets a entry for both directions of travel.  So an event, e.g. work zone, could be traveled in one direction but a road can be traveled in two directions.

Thanks

**Comment by sknick-iastate** [comment]
@DeraldDudley 

1. I think Maricopa County is a great example where this is valuable.  Maricopa County receives the data from the cities then compiles the data in a single location.  In this situation, Maricopa County would be the `feed_producer` and each of the cities would be the `event_issuingOrganization`

2. I'm fine with either but would like feedback from other.  My thought is that if it is verified then it is more accurate but either approach would work.  

As a note, I didn't include the milepost demarcation/accuracy/verified since they should have the same verification status as the begin/end.  For the begin/end points, they can be verified by a physical device (smart arrow board, connected pins/panels, etc).  I can't think of a way that you can verify a milepost.  I assumed the method of obtaining the milepost/reference post should be covered in the LRS portion of the [metadata](https://github.com/sknick-iastate/jpo-wzdx/blob/RelationalDataModel/data-tables/metadata.md#metadata)

File: data-tables/road_events.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+**event_subidentifier** | ID | A unique identifier issued by data feed provider that provides additional references to project or activity | Optional | This identifier may be used in more than one feed as a reference to an<br>agency project number or permit ID
+**geometry_type** | Enumeration: Multipoint or LineString|May be represented as a linestring or a multipoint as defined in the geoJson specification.|Required|
+**geometry**|Coordinate(s); Float|A coordinate pair or an array of coordinates. In either case, the first coordinate is the beginning point and the last coordinate is the ending point of the road event|Required|Coordinate pairs and coordinate arrays are formatted according to the geoJson spec
+**road_name**|Text|Publicly known name of the road on which the event occurs.|Required|
+**road_number**|Text|The road number designated by a jurisdiction such as a county, state or interstate|Optional|Examples I-5, VT 133
+**event_direction**|Text|The digitization direction of the road that is impacted by the event.|Required|Example North (for I-5 North)
```
Comment (chuehlien) [inline data-tables/road_events.md]: Is `event_direction` still an Enum field?

Comment (sknick-iastate) [reply data-tables/road_events.md]: Yes it is.  I made the correction.  The enumerations are already present in the [enumerations-fields file](https://github.com/sknick-iastate/jpo-wzdx/blob/RelationalDataModel/data-tables/enumerated-fields.md)

Thanks!

File: create-feed/README.md [hunk 2]
```diff
@@ -25,46 +25,68 @@ To interact directly with the DAVI Data Team and to get help implementing the sp
 ![Posting An Issue](https://github.com/usdot-jpo-ode/jpo-wzdx/blob/master/images/issues.png)
 
 ### Data File Types
-The WZDx v1.1 data feed will be specified as an XML file or a JSON file. Both JSON and XML have been chosen as the file types of choice because they are:
+The WZDx data feed will be specified as an XML file, a JSON file or a geoJSON file. All three have been chosen as the file types of choice because they are:
 - A lightweight data exchange format.
 - Easy for humans to read and write.
 - Easy for machines to parse and generate. 
 
 The file will contain one or more work zone activity entries. A work zone activity entry is defined as a description of work zone characteristics for a road segment along a single roadway in a single direction. The WorkZoneActivity data frame is composed of the [Common Core Data Dictionary elements (Table 1)](https://github.com/usdot-jpo-ode/jpo-wzdx/blob/master/data-tables/common-core-dictionary.md#table1.-common-core-data) and depicted in Figure 1.
```
Comment (CraigMoore-Sea) [inline create-feed/README.md]: Need to remove or change the reference to the Common Core Data Dictionary here.

Comment (sknick-iastate) [reply create-feed/README.md]: I removed the reference and pointed it to the road_events table that replaced the common core

**Comment by j-d-b** [comment]
Another general thought is that we don't need to prefix fields on the `road_event` table with `event`. Since they're in the `road_event` table, it's already clear what the field is describing. Currently it also feels inconsistent because not all fields are prefixed. For example, we have `road_event_id`, `event_subidentifier`, and `road_name` all on the same table.

**Comment by sknick-iastate** [comment]
> Quick one:
> 
> `event_issuingOrganization` should be `event_issuing_organization`

Corrected and also updated ERD diagram

**Comment by sknick-iastate** [comment]
I removed the prefix of event for all fields in the road_events table.  

I have also uploaded the [sample xml, json and xsd files](https://github.com/sknick-iastate/jpo-wzdx/tree/RelationalDataModel/sample-files).  I'll update the top post to reflect that those changes have been made.

**Comment by j-d-b** [comment]
Looking good; great progress. I have a couple final comments:

1. Since we decided to remove the `event` prefix in the `road_event` table, I feel like we should also remove the `feed` prefix from the fields in the `road_event_feed_info` table.
2. This PR changes V1.1's `reducedSpdPosted` to `reduced_speed_limit` which is minor and I like it better. Would you be open to changing `total_lanes` to `total_num_lanes` or `total_number_lanes`? I found it confusing that `open_lanes` and `closed_lanes` were both enumerations by `total_lanes` was an Integer; specifying in the field name would make this clear. I was planning to propose this in a new PR and I've been an advocate for all changes requiring a new PR but we're already slightly renaming a lot of fields here so I think it could be allowable in this context.
3. I want to suggest changing the `road_events` table `road_event_id` field name to `id`. Now that the `event` prefix is removed from all other fields it feels unnecessary here as well. It could be a good design pattern to have the primary key always called `id` within the table and then the foreign key in related tables uses the linking table name + `_id` (e.g. `road_event_id`). The same change would be made in the `road_event_feed_info` table.

**Comment by DeraldDudley** [comment]
> Looking good; great progress. I have a couple final comments:
> 
> 1. Since we decided to remove the `event` prefix in the `road_event` table, I feel like we should also remove the `feed` prefix from the fields in the `road_event_feed_info` table.
> 2. This PR changes V1.1's `reducedSpdPosted` to `reduced_speed_limit` which is minor and I like it better. Would you be open to changing `total_lanes` to `total_num_lanes` or `total_number_lanes`? I found it confusing that `open_lanes` and `closed_lanes` were both enumerations by `total_lanes` was an Integer; specifying in the field name would make this clear. I was planning to propose this in a new PR and I've been an advocate for all changes requiring a new PR but we're already slightly renaming a lot of fields here so I think it could be allowable addition in this scenario.
> 3. I want to suggest changing the `road_events` table `road_event_id` field name to `id`. Now that the `event` prefix is removed from all other fields it feels unnecessary here as well. It could be a good design pattern to have the primary key always called `id` within the table and then the foreign key in related tables uses the linking table name + `_id` (e.g. `road_event_id`). The same change would be made in the `road_event_feed_info` table. Let me know your thoughts.

RE Item 3: If i had to program against the feed, I would prefer consistent field names across the whole specification.

**Comment by sknick-iastate** [comment]
> Looking good; great progress. I have a couple final comments:
> 
> 1. Since we decided to remove the `event` prefix in the `road_event` table, I feel like we should also remove the `feed` prefix from the fields in the `road_event_feed_info` table.

This makes sense to me.  I'll go through and make the updates

> 2. This PR changes V1.1's `reducedSpdPosted` to `reduced_speed_limit` which is minor and I like it better. Would you be open to changing `total_lanes` to `total_num_lanes` or `total_number_lanes`? I found it confusing that `open_lanes` and `closed_lanes` were both enumerations by `total_lanes` was an Integer; specifying in the field name would make this clear. I was planning to propose this in a new PR and I've been an advocate for all changes requiring a new PR but we're already slightly renaming a lot of fields here so I think it could be allowable addition in this scenario.

This also makes sense and I will make the change.

> 3. I want to suggest changing the `road_events` table `road_event_id` field name to `id`. Now that the `event` prefix is removed from all other fields it feels unnecessary here as well. It could be a good design pattern to have the primary key always called `id` within the table and then the foreign key in related tables uses the linking table name + `_id` (e.g. `road_event_id`). The same change would be made in the `road_event_feed_info` table. Let me know your thoughts.

I would like to still keep this since it can help avoid the confusion of how to relate the tables.

**Comment by j-d-b** [comment]
> I would like to still keep this since it can help avoid the confusion of how to relate the tables.

> If i had to program against the feed, I would prefer consistent field names across the whole specification.

Gotcha, I didn't think of that aspect. Good points.

**Comment by sknick-iastate** [comment]
Thanks @j-d-b 

I think I got all of the changes to remove the feed prefix and change to `total_num_lanes`

**Comment by j-d-b** [comment]
Found something else, in the project root README, the "Common Core Data Dictionary" bullet is still in the Table of Contents and with a hyperlink that doesn't do anything.

**Comment by j-d-b** [comment]
> > I would like to still keep this since it can help avoid the confusion of how to relate the tables.
> 
> > If i had to program against the feed, I would prefer consistent field names across the whole specification.
> 
> Gotcha, I didn't think of that aspect. Good points.

To be consistent, should the the PK/ID field in the `road_event_feed_info` table be `road_event_feed_info_id` rather than `feed_info_id`?

**Comment by sknick-iastate** [comment]
I removed the common core reference.

I don't have a problem leaving the id as `feed_info_id`

**Comment by DeraldDudley** [comment]
I mapped  v1.1 to the PR #54 Model
I marked issues with **!!-** in the **PR #54 Model** column
Just  a few minor issues...
<ol>
<li>DataFrame.StartDateTime.startDateTime-cancelled - How are we handling cancelled events?
<li>DataFrame.StartDateTime.timeConfidenceLevel - Time Confidence Level is marked for future use.  Should we capture it in an issue? 
<li>Milepost Accuracy was dropped.  If were making a "clean" relational model that maps directly from version 1.1 to version  2.0 then dropping the milepost accuracy should be done through a pull request.
<li>Ambiguous field names - road_event_feed_info.update_date and road_event.update_date
<li>road_event_feed_info.publishers was created to normalize the road_event.issuing_organization.
<br> 2 options
<ol>
<li>rename road_event_feed_info.publishers to road_event_feed_info.issuing_organization and drop road_event.issuing_organization - or
<li>drop road_event_feed_info.publishers
<br>Option 5.ii is the "most" normalized
</ol>
</ol>

V1.1   Object | V1.1 Name | Action | V 2.0 Table | V 2.0 Field | PR #54 Model | New Fields
-- | -- | -- | -- | -- | -- | --
Common   Core | Identifier | Renames | road_events | road_event_id | road_event_id | feed_info_id
Common   Core | Subidentifier | Kept | road_events | subidentifier | subidentifier | geometry_type
Common   Core | DataFrame.StartDateTime.startDateTime-est | Split into start_date     start_date_accuracy | road_events | start_date | start_date |  
Common   Core | DataFrame.StartDateTime.startDateTime-ver | Split into start_date     start_date_accuracy | road_events | start_date_accuracy | start_date_accuracy |  
Common   Core | DataFrame.StartDateTime.startDateTime-cancelled | In the start date accuracy   enumeration? | road_events |   | **!! - How are cancellations covered?** |  
Common   Core | DataFrame.StartDateTime.timeConfidenceLevel | Dropped | road_events |   | **!! - Time confidence level was dropped.** |  
Common   Core | DataFrame.EndDateTime.endDateTime-est | Split into end_date     end_date_accuracy | road_events | end_date | end_date |  
Common   Core | DataFrame.EndDateTime.endDateTime-ver | Split into start_date     start_date_accuracy | road_events | end_date_accuracy | end_date_accuracy |  
Common   Core | DataFrame.EndDateTime.endDateTime-cancelled | In the end date accuracy   enumeration? | road_events |   | **!! - How are cancellations covered?** |  
Common   Core | DataFrame.BeginLocation.roadName | Reduced to one field for whole   event | road_events | road_name | road_name |  
Common   Core | DataFrame.BeginLocation.roadNum | Reduced to one field for whole   event | road_events | road_num | road_number |  
Common   Core | DataFrame.BeginLocation.roadDirection | Renamed to event_direction,        Reduced to one field for whole event | road_events | event_direction | event_direction |  
Common   Core | DataFrame.BeginLocation.latitude-est | Split into geometry      accuracy field | road_events | geometry (beginning coordinate) | geometry      (beginning and ending coordinates) |  
Common   Core | DataFrame.BeginLocation.latitude-ver | Split into geometry      accuracy field | road_events | beginning_accuracy | beginning_accuracy |  
Common   Core | DataFrame.BeginLocation.longitude-est | Split into geometry      accuracy field | road_events |   |   |  
Common   Core | DataFrame.BeginLocation.longitude-ver | Split into geometry      accuracy field | road_events |   |   |  
Common   Core | DataFrame.BeginLocation.milepost-est | Renamed beginning_milepost     milepost accuracy dropped | road_events | beginning_milepost | beginning_milepost |  
Common   Core | DataFrame.BeginLocation.milepost-ver | milepost accuracy dropped | road_events |   | **!! - milepost accuracy dropped** |  
Common   Core | DataFrame.BeginLocation.crossStreet | Renamed | road_events | beginning_cross_street | beginning_cross_street |  
Common   Core | DataFrame.EndLocation.roadName | Reduced to one field for whole   event | road_events |   |   |  
Common   Core | DataFrame.EndLocation.roadNum | Reduced to one field for whole   event | road_events |   |   |  
Common   Core | DataFrame.EndLocation.roadDirection | Renamed to event_direction,        Reduced to one field for whole event | road_events |   |   |  
Common   Core | DataFrame.EndLocation.latitude-est | Split into geometry and accuracy   field | road_events | geometry (ending coordinate) |   |  
Common   Core | DataFrame.EndLocation.latitude-ver | Split into geometry and accuracy   field | road_events | ending_accuracy | ending_accuracy |  
Common   Core | DataFrame.EndLocation.milepost-est | Split into ending_milepost        milepost accuracy dropped | road_events |   | ending_milepost |  
Common   Core | DataFrame.EndLocation.milepost-ver | milepost accuracy dropped | road_events |   | **!! - milepost accuracy dropped** |  
Common   Core | DataFrame.EndLocation.crossStreet | Renamed | road_events | ending_cross_street | ending_cross_street |  
Common   Core | Wz-Status | Renamed | road_events | event_status | event_status |  
Common   Core | totalLanes | Renamed | road_events |   | total_num_lanes |  
Common   Core | openLanes | Renamed | road_events |   | open_lanes |  
Common   Core | closedLanes | Renamed | road_events |   | closed_lanes |  
Common   Core | closedSholders | Renamed | road_events |   | closed_shoulders |  
Common   Core | workersPresent | Renamed | road_events | workers_present | workers_present |  
Common   Core | reducedSpdPosted | Renamed | road_events | reduced_speed_limit | reduced_speed_limit |  
Common   Core | RoadRestrictions | Renamed | road_events | restrictions | restrictions |  
Common   Core | Description | Kept | road_events | description | description |  
Common   Core | IssuingOrganization | Renamed | road_events | issuing_organization | issuing_organization |  
Common   Core | timestampEventCreation | Renamed | road_events | creation_date | creation_date |  
Common   Core | timestampEventUpdate | Renamed | road_events | update_date | **!!- update_date** |  
WZDx Header | timeStampUpdate | Renamed | road_event_feed_info | update_date | **!!- update_date** |  
WZDx Header | publisher | new - meant to replace issuing_organization from road_event_table | road_event_feed_info |   | **!! - Drop or Normalize?** |  
WZDx Header | metadataURL | Renamed | road_event_feed_info | metadata | metadata |  
WZDx Header | versionNo | Renamed | road_event_feed_info | version | version |

**Comment by sknick-iastate** [comment]
> I mapped v1.1 to the PR #54 Model
> I marked issues with **!!-** in the **PR #54 Model** column
> Just a few minor issues...
> 
> 1. DataFrame.StartDateTime.startDateTime-cancelled - How are we handling cancelled events?

I think this should be covered in the enumerated types on [event_status](https://github.com/sknick-iastate/jpo-wzdx/blob/RelationalDataModel/data-tables/enumerated-fields.md#enumerated-types) 

> 2. DataFrame.StartDateTime.timeConfidenceLevel - Time Confidence Level is marked for future use.  Should we capture it in an issue?

It would be good to submit an issue for this one.  There was hesitancy including it in v1.1 which is why it had the future use reference.

> 3. Milepost Accuracy was dropped.  If were making a "clean" relational model that maps directly from version 1.1 to version  2.0 then dropping the milepost accuracy should be done through a pull request.

This should be covered under the `beginning_accuracy` and `ending_accuracy`.  It is the same reason why we don't have a latitude accuracy and longitude accuracy field.

> 4. Ambiguous field names - road_event_feed_info.update_date and road_event.update_date

I think it may be good to change the road_event_feed_info.update_date to feed_update_date.

> 5. road_event_feed_info.publishers was created to normalize the road_event.issuing_organization.
>     2 options
>    
>    1. rename road_event_feed_info.publishers to road_event_feed_info.issuing_organization and drop road_event.issuing_organization - or2. drop road_event_feed_info.publishers
>       Option 5.ii is the "most" normalized

I think 5.ii makes more sense

**Comment by DeraldDudley** [comment]
> > I mapped v1.1 to the PR #54 Model
> > I marked issues with **!!-** in the **PR #54 Model** column
> > Just a few minor issues...
> > 
> > 1. DataFrame.StartDateTime.startDateTime-cancelled - How are we handling cancelled events?
> 
> I think this should be covered in the enumerated types on [event_status]
(https://github.com/sknick-iastate/jpo-wzdx/blob/RelationalDataModel/data-tables/enumerated-fields.md#enumerated-types)

DD:  Im good with that.

> > 1. DataFrame.StartDateTime.timeConfidenceLevel - Time Confidence Level is marked for future use.  Should we capture it in an issue?
> 
> It would be good to submit an issue for this one. There was hesitancy including it in v1.1 which is why it had the future use reference.

DD: I'll submit the issue to document it.

> > 1. Milepost Accuracy was dropped.  If were making a "clean" relational model that maps directly from version 1.1 to version  2.0 then dropping the milepost accuracy should be done through a pull request.
> 
> This should be covered under the `beginning_accuracy` and `ending_accuracy`. It is the same reason why we don't have a latitude accuracy and longitude accuracy field.

DD: I have 2 concerns... 
<ol>
<li>In geography, <a href="https://www.e-education.psu.edu/geog469/node/262">Positional Accuracy</a>, has a very specific meaning.  It is the degree to which information on a map or in a digital database matches true or accepted values and is <a href="https://pubs.usgs.gov/fs/1999/0171/report.pdf">measurable</a >.

Using the terms beginning_accuracy and ending_accuracy, in the context of spatial data, implies we are measuring positional accuracy;  We are not.

I suggest changing beginning_accuracy to beginning_is_verified and changing ending_accuracy to ending_is_verified so we do not convey the wrong meaning.

<li> The milepost_beginning and and the beginning coordinate are two different features.  The beginning_mile post refers to the nearest milepost to the beginning coordinate.  If milepost are dispersed at 1/10 of a mile intervals, the beginning_milepost could be as far as 528 feet away from the beginning coordinate.  Using beginning_accuracy to convey a level of confidence in the beginning_milepost is an inaccurate use of field.
</ol>
> > 1. Ambiguous field names - road_event_feed_info.update_date and road_event.update_date
> 
> I think it may be good to change the road_event_feed_info.update_date to feed_update_date.

DD: Thanks
 
> > 1. road_event_feed_info.publishers was created to normalize the road_event.issuing_organization.
> >    2 options
> >    
> >    1. rename road_event_feed_info.publishers to road_event_feed_info.issuing_organization and drop road_event.issuing_organization - or2. drop road_event_feed_info.publishers
> >       Option 5.ii is the "most" normalized
> 
> I think 5.ii makes more sense

DD: Thanks

**Comment by j-d-b** [comment]
@sknick-iastate I updated the XML schema as well as the sample feed files in  `sample-files` and the examples in `create-feed` and made a PR to your forked repo for review ([see here](https://github.com/sknick-iastate/jpo-wzdx/pull/2)).

**Comment by sknick-iastate** [comment]
> > > 1. Milepost Accuracy was dropped.  If were making a "clean" relational model that maps directly from version 1.1 to version  2.0 then dropping the milepost accuracy should be done through a pull request.
> > 
> > 
> > This should be covered under the `beginning_accuracy` and `ending_accuracy`. It is the same reason why we don't have a latitude accuracy and longitude accuracy field.
> 
> DD: I have 2 concerns...
> 
> 1. In geography, [Positional Accuracy](https://www.e-education.psu.edu/geog469/node/262), has a very specific meaning.  It is the degree to which information on a map or in a digital database matches true or accepted values and is [measurable](https://pubs.usgs.gov/fs/1999/0171/report.pdf).
>    Using the terms beginning_accuracy and ending_accuracy, in the context of spatial data, implies we are measuring positional accuracy;  We are not.
>    I suggest changing beginning_accuracy to beginning_is_verified and changing ending_accuracy to ending_is_verified so we do not convey the wrong meaning.

May I suggest we handle this in a separate PR?  Right now accuracy is also described in the [metadata](https://github.com/usdot-jpo-ode/jpo-wzdx/blob/master/data-tables/metadata.md) when talking about verification. So I think the field description aligns with the metadata but we can change both of those independently in a separate pull request.

> 2. The milepost_beginning and and the beginning coordinate are two different features.  The beginning_mile post refers to the nearest milepost to the beginning coordinate.  If milepost are dispersed at 1/10 of a mile intervals, the beginning_milepost could be as far as 264 feet away from the beginning coordinate.  Using beginning_accuracy to convey a level of confidence in the beginning_milepost is an inaccurate use of the field.
> 

I think this is open for interpretation on how it is adapted from v1.1.  My adaption is looking at the entire beginning and ending data frame and providing an accuracy/verification for that rather then on individual fields like you are suggesting.  

My adaption is based on an IOO perspective that it is all or nothing.  I don't see a situation where the coordinates are verified and the mile reference is estimated.  If we did not have a milepost then it is an optional field and it would not be provided. There is also information in the [metadata](https://github.com/usdot-jpo-ode/jpo-wzdx/blob/master/data-tables/metadata.md) on the LRS which should describe how the milepost are located. It also gets back to one of my previous points in how can you verify a milepost?

I don't feel this breaks from v1.1 since their are multiple different ways we could handle this.  I'm open to others suggestions and can make the change if others feel like we do need a verification/accuracy on milepost.

**Comment by DeraldDudley** [comment]
+1 Derald Dudley (USDOT)

**Comment by j-d-b** [comment]
+1 Jacob Brady (KCUS - MassDOT)

**Comment by DeraldDudley** [comment]
Have once.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by sknick-iastate** [comment]
+1

**Comment by SinclairStolle-IowaDOT** [comment]
yes

**Comment by mikedavidson-paturnpike** [comment]
Yes

**Comment by michelle315** [comment]
Yes

**Comment by DavidRush123** [comment]
Yes

**Comment by davidcraig4300** [comment]
Yes

**Comment by CraigMoore-Sea** [comment]
Yes

**Comment by sergebeaudry** [comment]
Yes

**Comment by sammyghorbanian** [comment]
yes - sammy on behalf of uber atg

**Comment by DrewClark-KY** [comment]
Yes - KYTC

**Comment by alexdemisch** [comment]
Yes (SFMTA)

**Comment by natedeshmukhtowery** [comment]
On behalf of Sam Krassenstein, City of Detroit - Yes

**Comment by NeilBoudreau** [comment]
Yes (MassDOT)  I was somewhat skeptical that we were broadening the category of "Work Zone Data" a bit to wide to now include and allow a more generic "road events" classification, but I can understand the desire of the others.

**Comment by GreyIC** [comment]
Yes (Jim Williams on behalf of INRIX)

**Comment by sergebeaudry** [comment]
Yes

**Pull request merged**